scipy/sparse/tests/test_construct.py

```diff
@@ -430,15 +430,15 @@ def test_bmat(self):
 
         with assert_raises(ValueError) as excinfo:
             construct.bmat([[A.tocsc()], [B.tocsc()]])
-        excinfo.match(r'Mismatching dimensions along axis 1: {1, 2}')
+        excinfo.match(r'Mismatching dimensions along axis 1: ({1, 2}|{2, 1})')
```

Comment:
For the reasoning behind this see [here](https://github.com/conda-forge/scipy-feedstock/pull/204#issuecomment-1194573592):
> The difference is [here](https://github.com/scipy/scipy/blob/b76137f0d4fd82bd74b2e51af33af3ece44e45bc/scipy/sparse/_construct.py#L453) where scipy does `set(b.shape[other_axis] for b in blocks)`. This boils down to `set([2, 1])` which on PyPy returns `{2, 1}` and on CPython `{1, 2}`. This seems like the test is being over-specific.